Mathlib/MeasureTheory/Order/UpperLower.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Kexing Ying
+-/
+import Mathlib.Analysis.Normed.Order.UpperLower
+import Mathlib.Logic.Lemmas
+import Mathlib.MeasureTheory.Covering.BesicovitchVectorSpace
+import Mathlib.Topology.Order.DenselyOrdered
+
+#align_import measure_theory.order.upper_lower from "leanprover-community/mathlib"@"b1abe23ae96fef89ad30d9f4362c307f72a55010"
+
+/-!
+# Order-connected sets are null-measurable
+
+This file proves that order-connected sets in `ℝⁿ` under the pointwise order are null-measurable.
+Recall that `x ≤ y` iff `∀ i, x i ≤ y i`, and `s` is order-connected iff
+`∀ x y ∈ s, ∀ z, x ≤ z → z ≤ y → z ∈ s`.
+
+## Main declarations
+
+* `Set.OrdConnected.null_frontier`: The frontier of an order-connected set in `ℝⁿ` has measure `0`.
+
+## Notes
+
+We prove null-measurability in `ℝⁿ` with the `∞`-metric, but this transfers directly to `ℝⁿ` with
+the Euclidean metric because they have the same measurable sets.
+
+Null-measurability can't be strengthened to measurability because any antichain (and in particular
+any subset of the antidiagonal `{(x, y) | x + y = 0}`) is order-connected.
+
+## TODO
+
+Generalize so that it also applies to `ℝ × ℝ`, for example.
+-/
+
+open Filter MeasureTheory Metric Set
+open scoped Topology
+
+namespace Set
+variable {ι : Type*} {α : ι → Type*} {i : ι} {s : Set ι} {t : ∀ i, Set (α i)}
+
+-- lemma eval_image_pi_of_not_mem (hi : i ∉ s) : Function.eval i '' s.pi t = univ :=
+--   eq_univ_of_forall (by simp)
+
+end Set
+
+section
+variable {α : Type*} [TopologicalSpace α] [Preorder α] {x : α}
+
+instance OrderDual.instNeBotNhdsWithinIoi [(𝓝[<] x).NeBot] : (𝓝[>] toDual x).NeBot := ‹_›
+instance OrderDual.instNeBotNhdsWithinIio [(𝓝[>] x).NeBot] : (𝓝[<] toDual x).NeBot := ‹_›
+
+end
+
+section
+variable {ι : Type*} {π : ι → Type*} [∀ i, TopologicalSpace (π i)] [∀ i, Preorder (π i)]
+open TopologicalSpace
+
+-- lemma isOpenMap_eval (i : ι) : IsOpenMap (Function.eval i : (∀ i, π i) → π i) := by
+--   refine (isTopologicalBasis_pi fun _ ↦ isTopologicalBasis_opens).isOpenMap_iff.2 ?_
+--   rintro _ ⟨U, s, hU, rfl⟩
+--   obtain h | h := ((s : Set ι).pi U).eq_empty_or_nonempty
+--   · simp [h]
+--   by_cases hi : i ∈ s
+--   · rw [eval_image_pi (mod_cast hi) h]
+--     exact hU _ hi
+--   · sorry
```

Comment:
To the topologists again, I didn't end needing this but do we really not have it?? We do have `isOpenMap_fst`/`isOpenMap_snd`.

Reply:
Moved to #13642